GAE/src/org/akvo/flow/api/app/DataPointServlet.java

```diff
@@ -83,21 +84,11 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
             if (device != null) {
                 log.info("Found device id: " + device.getKey().getId());
                 log.fine("Found device: " + device);
-                //Find which assignments we are part of
-                List<DataPointAssignment> assList =
-                        dataPointAssignmentDao.listByDeviceAndSurvey(device.getKey().getId(), dpReq.getSurveyId());
-                //Combine their point lists
-                Set<Long> pointSet = new HashSet<>();
-                for (DataPointAssignment ass : assList) {
-                    pointSet.addAll(ass.getDataPointIds());
-                }
-                //Fetch the data points
-                if (ALL_DATAPOINTS.equals(pointSet)) {
-                    dpList = surveyedLocaleDao.listLocalesBySurveyGroupId(dpReq.getSurveyId());
-                } else {
-                    List<Long> pointList = new ArrayList<>();
-                    pointList.addAll(pointSet);
-                    dpList = surveyedLocaleDao.listByKeys(pointList);
+                List<SurveyedLocale> dpList = getDataPointList(device.getKey().getId(), dpReq.getSurveyId());
+                if (dpList == null) {
+                    res.setCode(String.valueOf(HttpServletResponse.SC_BAD_REQUEST));
```

Comment:
Perhaps we should return a `404` instead? @valllllll2000 how would the device behave when we return one of these error codes?